### PR TITLE
refactoring w.r.t new abi

### PR DESCRIPTION
**Pull request by jkrivine**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,5 +21,8 @@
     "dependencies": {
         "pinst": "^2.1.6"
     },
-    "packageManager": "yarn@3.1.0"
+    "packageManager": "yarn@3.1.0",
+    "resolutions": {
+        "@mangrovedao/mangrove-core": "portal:/Users/jeankrivine/Documents/git/mangrove-core"
```
Comment (lnist) [inline package.json:26]: remember to revert this - ideally it should not have been pushed :)

File: packages/mangrove.js/src/liquidityProvider.ts [hunk 2]
```diff
@@ -295,29 +260,17 @@ class LiquidityProvider {
...
-        inbound_tkn.address,
-        inbound_tkn.toUnits(wants),
-        outbound_tkn.toUnits(gives),
-        gasreq ? gasreq : 0,
-        gasprice ? gasprice : 0,
-        pivot ?? 0,
-        this.#optValueToPayableOverride(overrides, fund)
-      );
-    }
+    let poster = this.logic ? this.logic : this.mgv;
```
Comment (lnist) [inline packages/mangrove.js/src/liquidityProvider.ts:263]: add .contract here like other places?

File: packages/mangrove.js/src/liquidityProvider.ts [hunk 3]
```diff
@@ -553,30 +444,27 @@ class LiquidityProvider {
     const gasreq = opts.gasreq ? opts.gasreq : await this.#gasreq();
     const gasprice = opts.gasprice ? opts.gasprice : 0;
     // this computes the total provision required for a new offer on the market
-    const bounty = await this.market.getOfferProvision(ba, gasreq, gasprice);
+    const provision = await this.market.getOfferProvision(ba, gasreq, gasprice);
+    let lockedProvision: Bigish;
     // checking now the funds that are either locked in the offer or on the maker balance on Mangrove
-    let lockedProvision = Big(0);
     if (opts.id) {
-      const offer = await this.market.offerInfo(ba, opts.id);
```
Comment (lnist) [inline packages/mangrove.js/src/liquidityProvider.ts:560]: Can we take this over coffee? :)

**Review by lnist — CHANGES_REQUESTED** [review]
I really like the simplifications!

File: packages/mangrove.js/src/liquidityProvider.ts (hunk 3 above)
Comment (jkrivine) [reply packages/mangrove.js/src/liquidityProvider.ts:560]: yes. I am just utilising here the function `provisionOf` that is Forwarder/Direct compatible

File: packages/mangrove.js/src/util/trade.ts [hunk 4]
```diff
@@ -173,87 +178,40 @@ class Trade {
               },
             ],
             fillWants: fillWants,
-            ba: "asks",
+            ba: bs === "buy" ? "asks" : "bids",
```
Comment (lnist) [inline packages/mangrove.js/src/util/trade.ts:181]: I've added a BsToBa function you can use for this

File: packages/mangrove.js/src/util/trade.ts [hunk 5]
```diff
@@ -438,6 +396,7 @@ class Trade {
         ? [market.base, market.quote]
         : [market.quote, market.base];
 
+    await market.consoleAsks(["wants", "gives"]);
```
Comment (lnist) [inline packages/mangrove.js/src/util/trade.ts:399]: this should not be pushed

File: packages/mangrove.js/test/unit/mangrove.unit.test.ts [hunk 6]
```diff
@@ -21,7 +21,7 @@ describe("Mangrove functionality", () => {
       // start server
       const server = await node({
         ...defaultServerParams,
-        port: 8546,
+        port: 8545,
       }).connect();
```
Comment (lnist) [inline packages/mangrove.js/test/unit/mangrove.unit.test.ts:25]: why this change? 8544 would probably be even better to not clash with other tests.

File: packages/mangrove.js/src/util/trade.ts [hunk 7]
```diff
@@ -140,23 +140,28 @@ class Trade {
    * market.buy({volume: 100, price: '1.01'}) //use strings to be exact
    * ```
    */
-  buy(
+  order(
```
Comment (lnist) [inline packages/mangrove.js/src/util/trade.ts:143]: In the past there was some switching gives and wants around depending on buy/sell. I assume all that is still taken care of. It is great we got this simplified!

**Review by lnist — CHANGES_REQUESTED** [review]

File: packages/mangrove.js/src/util/trade.ts (hunk 7 above)
Comment (jkrivine) [reply packages/mangrove.js/src/util/trade.ts:143]: This is taken care line 151

File: packages/mangrove.js/src/market.ts [hunk 8]
```diff
@@ -70,20 +70,18 @@ namespace Market {
 
   export type TradeParams = {
     slippage?: number;
-  } & ({ mangroveOrder?: MangroveOrderParams } | { offerId?: number }) &
+    fillOrKill?: boolean;
```
Comment (peterMangrove) [inline packages/mangrove.js/src/market.ts:73]: Does it make sense, that you now can both do a FoK and restingorder?

**Review by lnist — APPROVED** [review]

File: packages/mangrove.js/src/market.ts (hunk 8 above)
Comment (jkrivine) [reply packages/mangrove.js/src/market.ts:73]: At least this has an interpretation by the backend: revert if partialFill & resting order failed to be posted

**Pull request merged**